airflow/providers/cncf/kubernetes/pod_template_file_examples/git_sync_template.yaml

```diff
@@ -26,21 +26,21 @@ metadata:
 spec:
   initContainers:
     - name: git-sync
-      image: "registry.k8s.io/git-sync/git-sync:v3.6.3"
+      image: "registry.k8s.io/git-sync/git-sync:v4.0.0"
       env:
-        - name: GIT_SYNC_BRANCH
+        - name: GITSYNC_REF
```

Comment:
As much as I don't want to say it, I think we need to set both the old and new styles. If folks have existing <v4 images mirrored this will break.

Reply:
ok. As I can see, git-sync v4.0.0+ accept `GIT_SYNC_BRANCH` and `GIT_SYNC_REV` flags, they just deprecated, so I guess we can still use them for a some time

Reply:
I also add example of git-sync v4.0.0+ usage in pod_template in separate yaml file

Reply:
> git-sync v4.0.0+ accept GIT_SYNC_BRANCH and GIT_SYNC_REV flags, they just deprecated

Will we have deprecation warning if we provide both old and new styles to git-sync v4.0.0+ image?